susemanager-utils/susemanager-sls/salt/util/mgr_switch_to_venv_minion.sls

```diff
@@ -56,20 +56,32 @@ mgr_copy_salt_minion_keys:
     - unless:
       - test -f /etc/venv-salt-minion/pki/minion/minion_master.pub
 
+{%- if grains['transactional']==True %}
+mgr_enable_venv_salt_minion:
+    cmd.run:
+    - name: systemctl enable venv-salt-minion
+    - require:
+      - cmd: mgr_copy_salt_minion_keys
+{%- else %}
 mgr_enable_venv_salt_minion:
   service.running:
     - name: venv-salt-minion
     - enable: True
-    - require:
-      - cmd: mgr_copy_salt_minion_keys
+{%- endif %}
 
+{%- if grains['transactional']==True %}
 mgr_disable_salt_minion:
-  service.dead:
-    - name: salt-minion
-    - enable: False
+    cmd.run:
+    - name: systemctl disable salt-minion
     - require:
       - service: mgr_enable_venv_salt_minion
       - sls: services.salt-minion
+{%- else %}
+mgr_disable_salt_minion:
+  service.dead:
+    - name: salt-minion
+    - enable: False
```

Comment:
This state also requires `service: mgr_enable_venv_salt_minion` and `sls: services.salt-minion`